morphic.js

```diff
@@ -7818,10 +7824,40 @@ MenuMorph.prototype.select = function (aMenuItem) {
     this.selection = aMenuItem;
 };
 
+MenuMorph.prototype.mouseLeave = function () {
+    if (this.parent instanceof MenuMorph) {
+        // this is a submenu so we vanish when lose focus
+		this.destroy();
+    }
+}
+            
+
 MenuMorph.prototype.destroy = function () {
+var myself = this,
+    subitems = function(menu) {
+        return menu.children.filter(function (child) {
+                    return (child instanceof MenuItemMorph) &&
+                           (child.doubleClickAction instanceof MenuMorph);
+			} )
```

Comment:
Can't we just start over and rewrite Snap<i>!</i> in Scheme?  When you guys can't agree about the correct syntax there's something wrong with the underlying syntax spec.